### PR TITLE
Create Source Delete page

### LINKED ISSUE #804: Cry for help: deleting one of "My Sources" is not obvious (in fact... I can't figure it out)
As an editor on Cantus, I can "Add new sources," and add chants to those sources. This function works, because I have successfully done so. However, I now can't make my source go away.

I can delete individual chants (see below), but I can't find where I can delete the entire source.

Deleting a chant

So I guess there are two possible issues here:
1. If it's actually not possible to delete a source once you've created it, it'd be great for that to be possible.
2. If it is possible to delete a source, then perhaps the "Delete" button should be somewhere more obvious to dumb users like me.

### PR DESCRIPTION
**Pull request by jacobdgm**
This PR implements a couple of things related to the Source Delete page:
- Implements a new Source Delete page (i.e. fixes #804)
- On the Source Edit page, adds a link to the Source Delete page
- Makes some tweaks to the Chant Delete page, cleaning it up and making it resemble the new Source Delete page.
- Updates the path for the Chant Delete page, from `/chant-delete/<id>` to `/chant/<id>/delete/` (the Source Delete path is constructed similarly).
- Adds some tests for the Source Delete page, updates some to match the new Chant Delete path, and refactors some to follow best practices (i.e. using `reverse` rather than hard-coding paths)

### CODE REVIEW

**Comment by jacobdgm** [comment]
In addition to fixing #804, this PR brings us a step closer to resolving #486 - all we need now is a Sequence Delete page modelled on our existing Chant Delete page

File: django/cantusdb_project/main_app/views/source.py [hunk 1]
```diff
@@ -218,6 +224,40 @@ def form_valid(self, form):
...
+    """
+
+    model = Source
+    template_name = "source_confirm_delete.html"
+
+    def test_func(self):
+        user = self.request.user
+        source_id = self.kwargs.get(self.pk_url_kwarg)
+        source = get_object_or_404(Source, id=source_id)
+
```
Comment (lucasmarchd01) [inline django/cantusdb_project/main_app/views/source.py]: You could just do `return user_can_edit_source(user, source)` here although the function name doesn't exactly match.

File: django/cantusdb_project/main_app/views/source.py [hunk 2]
```diff
@@ -268,7 +308,7 @@ def form_valid(self, form):
         source = form.save()
 
         for old_editor in old_current_editors:
-            old_editor.sources_user_can_edit.remove(source)
+            old_editor.sources_3.remove(source)
```
Comment (lucasmarchd01) [inline django/cantusdb_project/main_app/views/source.py]: what is `sources_3`?

Comment (jacobdgm) [reply django/cantusdb_project/main_app/views/source.py]: oh! I don't recall making this change - something must have happened while sorting out the merge conflict. Thanks for catching this!

File: django/cantusdb_project/main_app/views/source.py (hunk 1 above)
Comment (jacobdgm) [reply django/cantusdb_project/main_app/views/source.py]: good! I made this substitution, and the tests still pass.

**Review by lucasmarchd01 — APPROVED** [review]

**Pull request merged**